BrainPortal/app/controllers/nh_loris_hooks_controller.rb

```diff
@@ -64,24 +64,70 @@ def file_list_maker
     s_group = Group.where_id_or_name(source_group_id).first if source_group_id
     base = Userfile.where(nil) # As seen in userfiles_controller
     base = Userfile.restrict_access_on_query(current_user, base, :access_requested => :read)
+
```

Comment:
This entire block should be moved into a new private method.

It should only return the bare information about the paths as a hash: e.g. { 'sub-1234' => [ 'anat/blah1_t1.nii.gz', 'anat/blah2_t1.nii.gz' ], 'sub-2345' => [], 'sub-33221' => [...] }